ctapipe/io/magiceventsource.py

```diff
@@ -0,0 +1,1337 @@
+            event_data['arrival_time'].append(arrival_time)
+            event_data['trigger_pattern'] = np.concatenate((event_data['trigger_pattern'], trigger_pattern))
+            event_data['stereo_event_number'] = np.concatenate((event_data['stereo_event_number'], stereo_event_number))
+            event_data['pointing_zd'] = np.concatenate((event_data['pointing_zd'], pointing_zd))
+            event_data['pointing_az'] = np.concatenate((event_data['pointing_az'], pointing_az))
+            event_data['pointing_ra'] = np.concatenate((event_data['pointing_ra'], pointing_ra))
+            event_data['pointing_dec'] = np.concatenate((event_data['pointing_dec'], pointing_dec))
+
+            event_data['MJD'] = np.concatenate((event_data['MJD'], mjd))
+
+            event_data['file_edges'].append(len(event_data['trigger_pattern']))
+
+        return event_data
+
+    def _find_pedestal_events(self):
+        """
+        This internal method identifies the IDs (order numbers) of the
+        pedestal events in the run.
+
+        Returns
+        -------
+        dict:
+            A dictionary of pedestal event IDs in M1/2 separately.
+        """
+
+        pedestal_ids = dict()
+
+        for telescope in self.event_data:
+            ped_triggers = np.where(self.event_data[telescope]['trigger_pattern'] == 8)
+            pedestal_ids[telescope] = ped_triggers[0]
+
+        return pedestal_ids
+
+    def _find_stereo_events(self):
+        """
+        This internal methods identifies stereo events in the run.
+
+        Returns
+        -------
+        list:
+            A list of pairs (M1_id, M2_id) corresponding to stereo events in the run.
+        """
+
+        m2_data_condition = (self.event_data['M2']['trigger_pattern'] == 128)
+
+        stereo_ids = []
+        n_m1_events = len(self.event_data['M1']['stereo_event_number'])
+
+        for m1_id in range(0, n_m1_events):
+            if self.event_data['M1']['trigger_pattern'][m1_id] == 128:
+                m2_stereo_condition = (self.event_data['M2']['stereo_event_number'] ==
+                                       self.event_data['M1']['stereo_event_number'][m1_id])
+
+                m12_match = np.where(m2_data_condition & m2_stereo_condition)
+
+                if len(m12_match[0]) > 0:
+                    stereo_pair = (m1_id, m12_match[0][0])
+                    stereo_ids.append(stereo_pair)
+
+        return stereo_ids
+
+    def _find_mono_events(self):
+        """
+        This internal method identifies the IDs (order numbers) of the
+        pedestal events in the run.
+
+        Returns
+        -------
+        dict:
+            A dictionary of pedestal event IDs in M1/2 separately.
+        """
+
+        mono_ids = dict()
+        mono_ids['M1'] = []
+        mono_ids['M2'] = []
+
+        m1_data_condition = self.event_data['M1']['trigger_pattern'] == 128
+        m2_data_condition = self.event_data['M2']['trigger_pattern'] == 128
```

Comment:
The "magic number" `128` is now repeated here for the 4th or 5th time ... I think it might be good to replace them by a nicely named costant.

Reply:
Fixed.